src/smc-webapp/course/handouts_panel.cjsx

```diff
@@ -0,0 +1,563 @@
+        for x in v
+            user = @props.user_map.get(x.account_id)
+            if user?
+                x.first_name = user.get('first_name')
+                x.last_name  = user.get('last_name')
+                x.name = x.first_name + ' ' + x.last_name
+                x.sort = (x.last_name + ' ' + x.first_name).toLowerCase()
+            else if x.email_address?
+                x.name = x.sort = x.email_address.toLowerCase()
+
+        v.sort (a,b) ->
+            return misc.cmp(a.sort, b.sort)
+
+        for x in v
+            @render_student_info(x.student_id, x)
+
+    render_student_info : (id, student) ->
+        <StudentHandoutInfo
+            key = {id}
+            actions = {@props.actions}
+            info = {@props.store.student_handout_info(id, @props.handout)}
+            title = {misc.trunc_middle(@props.store.get_student_name(id), 40)}
+            student = {id}
+            handout = {@props.handout}
+        />
+
+    render : ->
+        <div>
+            <StudentHandoutInfoHeader
+                key        = 'header'
+                title      = "Student"
+            />
+            {@render_students()}
+        </div>
+
+StudentHandoutInfoHeader = rclass
+    displayName : "CourseEditor-StudentHandoutInfoHeader"
+
+    propTypes :
+        title      : rtypes.string.isRequired
+
+    render_col: (step_number, key, width) ->
+        switch key
+            when 'last_handout'
+                title = 'Distribute to Student'
+                tip   = 'This column gives the status whether a handout was received by a student and lets you copy the handout to one student at a time.'
+        <Col md={width} key={key}>
+            <Tip title={title} tip={tip}>
+                <b>{step_number}. {title}</b>
+            </Tip>
+        </Col>
+
+
+    render_headers: ->
+        w = 12
+        <Row>
+            {@render_col(1, 'last_handout', w)}
+        </Row>
+
+    render : ->
+        <Row style={borderBottom:'2px solid #aaa'} >
+            <Col md=2 key='title'>
+                <Tip title={@props.title} tip={if @props.title=="Handout" then "This column gives the directory name of the handout." else "This column gives the name of the student."}>
+                    <b>{@props.title}</b>
+                </Tip>
+            </Col>
+            <Col md=10 key="rest">
+                {@render_headers()}
+            </Col>
+        </Row>
+
+StudentHandoutInfo = rclass
+    displayName : "CourseEditor-StudentHandoutInfo"
+
+    propTypes :
+        actions    : rtypes.object.isRequired
+        info       : rtypes.object.isRequired
+        title      : rtypes.oneOfType([rtypes.string,rtypes.object]).isRequired
+        student    : rtypes.oneOfType([rtypes.string,rtypes.object]).isRequired # required string (student_id) or student immutable js object
+        handout    : rtypes.oneOfType([rtypes.string,rtypes.object]).isRequired # required string (handout_id) or handout immutable js object
+
```

Comment:
Maintains similar construction to StudentAssignmentInfo yet still separate (again).